### PR TITLE
APi

### PR DESCRIPTION
**Pull request by sanjeev-cs**

### CODE REVIEW

File: backend/routes/auth/tokens.js [hunk 1]
```diff
@@ -13,9 +13,9 @@
     try {
       const userIP = req.ip;
       // Create a token with user IP
-      // console.log(process.env.SECRET_KET);
-      let key = process.env.SECRET_KET;
-      let expiresIn = 60 * 60 * 24; // 24 hours in seconds
```
Comment (ConnorBP) [inline backend/routes/auth/tokens.js:18]: Please put back that comment it's important

File: backend/routes/gallery/gallery.js [hunk 2]
```diff
@@ -1,56 +1,56 @@
...
+    // code here
 
     // Save canvas data to the database
     const result = await saveCanvasData(
       {
-        // manually destructured for security purposes
-        name: canvasData.name,
-        description: canvasData.description,
-        pixels: canvasData.pixels,
-        width: canvasData.width,
```
Comment (ConnorBP) [inline backend/routes/gallery/gallery.js:32]: Looks like you overwrote my changes with an old copy of the repo. Please fix.

**Review by ConnorBP — CHANGES_REQUESTED** [review]

File: backend/routes/auth/tokens.js (hunk 1 above)
Comment (sanjeev-cs) [reply backend/routes/auth/tokens.js:18]: OK

File: backend/routes/gallery/gallery.js (hunk 2 above)
Comment (sanjeev-cs) [reply backend/routes/gallery/gallery.js:32]: Sure

File: backend/routes/auth/tokens.js (hunk 1 above)
Comment (sanjeev-cs) [reply backend/routes/auth/tokens.js:18]: Done

File: backend/routes/gallery/gallery.js (hunk 2 above)
Comment (sanjeev-cs) [reply backend/routes/gallery/gallery.js:32]: Done

File: backend/routes/gallery/canvas.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ export const connectToDB = async () => {
     try {
         await client.connect();
         console.log("Connected to database");
-        return client.db(dbName); // Return the connected database instance
```
Comment (ConnorBP) [inline backend/routes/gallery/canvas.js:13]: No point removing these comments

File: backend/routes/gallery/canvas.js [hunk 4]
```diff
@@ -32,23 +32,19 @@ export const saveCanvasData = async (canvasData) => {
         console.error("Error inserting canvas data:", e.stack || e);
         throw new Error("Failed to insert canvas data");
     } finally {
-        if (db) db.client.close(); // Close the database connection
```
Comment (ConnorBP) [inline backend/routes/gallery/canvas.js:35]: This too

File: backend/routes/gallery/canvas.js [hunk 5]
```diff
@@ -32,23 +32,19 @@ export const saveCanvasData = async (canvasData) => {
...
+// Get all canvases collection
 export const getAllCanvases = async () => {
     let db;
     try {
         db = await connectToDB();
-        const collection = db.collection(collectionName);
-
-        // Retrieve all canvases from the collection
-        const canvases = await collection.find().toArray();
-        return canvases; // Return the array of canvases
```
Comment (ConnorBP) [inline backend/routes/gallery/canvas.js:47]: Put these comments back too

File: backend/routes/gallery/canvas.js [hunk 6]
```diff
@@ -32,23 +32,19 @@ export const saveCanvasData = async (canvasData) => {
...
+        const collection = db.collection(collectionName); 
+        return collection; 
     } catch (e) {
         console.error("Error retrieving canvases:", e.stack || e);
         throw new Error("Failed to retrieve canvases");
-    } finally {
-        if (db) db.client.close();
```
Comment (ConnorBP) [inline backend/routes/gallery/canvas.js:52]: This is an error it leaks database connections. Put it back and add a comment

**Review by ConnorBP — CHANGES_REQUESTED** [review]

File: backend/routes/gallery/canvas.js (hunk 3 above)
Comment (sanjeev-cs) [reply backend/routes/gallery/canvas.js:13]: Done

File: backend/routes/gallery/canvas.js (hunk 4 above)
Comment (sanjeev-cs) [reply backend/routes/gallery/canvas.js:35]: Done

File: backend/routes/gallery/canvas.js (hunk 5 above)
Comment (sanjeev-cs) [reply backend/routes/gallery/canvas.js:47]: Done

File: backend/routes/gallery/canvas.js (hunk 6 above)
Comment (ConnorBP) [reply backend/routes/gallery/canvas.js:52]: It's still missing

Comment (sanjeev-cs) [reply backend/routes/gallery/canvas.js:52]: I moved that logic in paginatedResults.js

**Review by ConnorBP — APPROVED** [review]

**Pull request merged**